### ISSUE TITLE
Midi clock filtered too much, slow reaction time to tempo changes

### ISSUE THREAD

**Issue by mattpepin** [moddevices/mod-host#36]
I use a Kemper Profiler guitar amp to generate the midi clock signal. However, when I switch to a different performance (bank/slot) with a different tempo, mod-host takes too long to adjust to the new tempo (it ramps up or down too slowly). I find that the filter does not react fast enough. It makes it impossible to use the midi clock for songs that change suddenly change tempo. If I use:

#define AVERAGE_FILTER_STEPS  7  (instead of 47)

and completely disable the binomial filter, it works perfectly.

**Comment by mattpepin** [comment]
I forgot to mention that I use mod-host to drive an LV2 metronome for the whole band, so timing is crucial ;)

**Comment by falkTX** [comment]
The MIDI Clock sync option has a very long jitter compensation filter because of a kernel issue in the MOD Duo X, which has since been corrected.
But I did not investigate if/how we can change this to a sane values, and what those values are.

How did you come up with "7", is it arbitrary?

**Comment by mattpepin** [comment]
Yes, it's totally arbitrary but I came up with it after some tests. In my case, it seems to work well.
